test/tests/feedTest.js

```diff
@@ -0,0 +1,156 @@
+				return Zotero.Promise.resolve();
+			}));
+		});
+		
+		it("should save a new feed to the feed library", function *() {
+			let feed = new Zotero.Feed();
+			feed.name = name;
+			feed.url = url;
+			
+			let id = yield feed.saveTx({skipEditCheck: true});
+			feed = yield Zotero.Feeds.getAsync(id);
+			feeds.push(feed);
+			
+			assert.equal(feed.name, name, "name is correct");
+			assert.equal(feed.url, url, "url is correct");
+			assert.equal(feed.libraryID, Zotero.Libraries.feedLibraryID, "library ID is correct");
+			assert.equal(feed.id, id, "ID is available");
+		});
+		it("should require library edit check override to save", function() {
+			let feed = new Zotero.Feed();
+			feed.url = url;
+			feed.name = name;
+			return assert.isRejected(feed.saveTx(), /^Cannot edit feed in read-only Zotero library$/);
+		});
+		it("should throw if name or url are missing", function *() {
+			let feed = new Zotero.Feed();
+			yield assert.isRejected(feed.saveTx({skipEditCheck: true}, /^Feed URL is not set$/));
+			
+			feed.name = name;
+			yield assert.isRejected(feed.saveTx({skipEditCheck: true}, /^Feed URL is not set$/));
+			
+			feed = new Zotero.Feed();
+			feed.url = url;
+			yield assert.isRejected(feed.saveTx({skipEditCheck: true}, /^Feed name is empty$/));
+		});
+		it("should throw when trying to save with an invalid URL", function() {
+			let feed = new Zotero.Feed();
+			feed.name = name;
+			feed.url = 'foo';
+			
+			return assert.isRejected(feed.saveTx({skipEditCheck: true}));
+		});
+		it("should save a feed item with all fields set", function *() {
+			let feed = new Zotero.Feed(),
+				refreshInterval = 30,
+				cleanupAfter = 2;
+			
+			feed.name = name;
+			feed.url = url;
+			feed.refreshInterval = refreshInterval;
+			feed.cleanupAfter = cleanupAfter;
+			
+			let id = yield feed.saveTx({skipEditCheck: true});
+			feed = yield Zotero.Feeds.getAsync(id);
+			feeds.push(feed);
+			
+			assert.equal(feed.name, name, "name is correct");
+			assert.equal(feed.url, url, "url is correct");
+			assert.equal(feed.refreshInterval, refreshInterval, "refreshInterval is correct");
+			assert.equal(feed.cleanupAfter, cleanupAfter, "cleanupAfter is correct");
+			
+			assert.equal(feed.libraryID, Zotero.Libraries.feedLibraryID, "library ID is correct");
+			assert.equal(feed.id, id, "ID is available");
+			assert.isUndefined(feed.lastCheck, "lastCheck is null");
+			assert.isUndefined(feed.lastUpdate, "lastUpdate is null");
+			assert.isUndefined(feed.lastCheckError, "lastCheckError is null");
+		});
+		it("should not allow setting a feed as child of another feed", function *() {
+			let feed = new Zotero.Feed();
+			
+			assert.throws(() => feed.parentID = 1, 'Cannot set parent key for feed', 'cannot set parent by ID');
+			assert.throws(() => feed.parentKey = 1, 'Cannot set parent key for feed', 'cannot set parent by key');
+		});
+		it("should not allow saving a feed with the same url", function *() {
```

Comment:
I'm not 100% set on this one, but I also don't see any reason to allow it. Interestingly, the test actually currently fails, even though the SQLite schema specifies UNIQUE for url.